src/utils/wallet.js

```diff
@@ -109,21 +162,83 @@ class Wallet {
         localStorage.setItem(KEY_WALLET_ACCOUNTS, JSON.stringify(this.accounts))
     }
 
+    getAccountId() {
+        const tempAccount = getTempAccount()
+        // temp account was set during create account
+        if (tempAccount && tempAccount.accountId) {
+            return tempAccount.accountId
+        } else {
+            return this.accountId
+        }
+    }
+
     selectAccount(accountId) {
         if (!(accountId in this.accounts)) {
             return false
         }
         this.accountId = accountId
         this.save()
     }
-
+    
     isLegitAccountId(accountId) {
         return ACCOUNT_ID_REGEX.test(accountId)
     }
-
+    
     async sendMoney(receiverId, amount) {
-        await this.getAccount(this.accountId).sendMoney(receiverId, amount)
+        const { accountId } = this
+        const { account, has2fa } = await this.getAccountAndState(accountId)
+        if (has2fa) {
+            const request = {
+                receiver_id: receiverId,
+                actions: [{ type: 'Transfer', amount }]
+            }
+            await this.makeTwoFactorRequest(account, request)
+        } else {
+            await account.sendMoney(receiverId, amount)
+        }
+    }
+    /********************************
+    Multisig Requests
```

Comment:
See above regarding moving 2FA stuff into separate module.

Reply:
in progress